maintenance/health/servicehealthcheck/healthchecker.go

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+// Created at 2019/10/18 Charlotte Pröller
+
+package servicehealthcheck
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/pace/bricks/maintenance/log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+type HealthCheck interface {
+	Name() string
+	InitHealthCheck() error
+	HealthCheck(currTime time.Time) (bool, error)
+	CleanUp() error
+}
+
+// ConnectionState can be used for Health Checks
+// It offers a Mutex and the Date of the last check for caching the result
+type ConnectionState struct {
+	LastCheck time.Time
+	isHealthy bool
+	err       error
+	m         sync.Mutex
+}
+
+type handler struct{}
+
+// checks map with all health checks, key: Name of the check
+var checks = make(map[string]HealthCheck)
+
+// initErrors map with all errors that happened in the initialisation of the health checks
+var initErrors = make(map[string]error)
+var router *mux.Router
+
+func (cs *ConnectionState) setConnectionState(healthy bool, err error, mom time.Time) {
+	cs.m.Lock()
+	defer cs.m.Unlock()
+	cs.isHealthy = healthy
+	cs.err = err
+	cs.LastCheck = mom
+}
+
+func (cs *ConnectionState) SetErrorState(err error, mom time.Time) {
+	cs.setConnectionState(false, err, mom)
```

Comment:
```suggestion
	cs.setConnectionState(err == nil, err, mom)
```

For example in the redis health check you do this:

```go
errRead := h.client.Get(cfg.HealthKey).Err()
h.State.SetErrorState(errRead, currTime)
```